apps/chat/src/components/ToolsetEditor/EditorForm/AuthField.tsx

```diff
@@ -0,0 +1,275 @@
+  type,
+  isSelected,
+  onClick,
+  onLogout,
+  onLogin,
+}: AuthTypeSectionProps) => {
+  const { t } = useTranslation(Translation.Common);
+
+  const toolsetDetails = useAppSelector(ToolsetSelectors.selectToolsetDetails);
+
+  const [withLogin, setWithLogin] = useState(WithLogin.WithLogin);
+  const isWithLogin = withLogin === WithLogin.WithLogin;
+  const isSignedIn = toolsetDetails && isToolsetSignedIn(toolsetDetails);
+
+  const handleWithLoginChange = (e: ChangeEvent<HTMLInputElement>) => {
+    setWithLogin(e.currentTarget.value as WithLogin);
+  };
+
+  const { Icon, name } = authTypeOptions[type];
+
+  const handleOnClick = useCallback(() => {
+    if (!isSignedIn) onClick(type);
+  }, [isSignedIn, onClick, type]);
+
+  return (
+    <div className="overflow-hidden rounded bg-layer-3">
+      <div
+        onClick={handleOnClick}
+        className={classNames(
+          'flex gap-3 border-l p-4',
+          isSelected ? 'border-accent-primary' : 'border-transparent',
+          !isSelected && !isSignedIn && 'cursor-pointer',
+        )}
+      >
+        <Icon
+          size={18}
+          className={classNames(
+            isSelected ? 'text-accent-primary' : 'text-secondary',
+          )}
+        />
+
+        <span
+          className={classNames(
+            'text-sm font-semibold',
+            isSelected ? 'text-accent-primary' : 'text-primary',
+          )}
+        >
+          {name}
+        </span>
+      </div>
+
+      {isSelected && type !== ToolsetAuthTypes.NONE && (
+        <div className="grid grid-cols-2 border-t border-tertiary">
+          <div className="flex flex-col gap-4 p-4">
+            <RadioButton
+              id={WithLogin.WithLogin}
+              name="with-auth"
+              caption={t(WithLogin.WithLogin)}
+              onChange={handleWithLoginChange}
+              value={WithLogin.WithLogin}
+              checked={isWithLogin}
+            />
+
+            <ToolsetLoginForm
+              onLogin={onLogin}
+              type={type}
+              toolset={toolsetDetails}
+              disabled={!isWithLogin}
+              onLogout={onLogout}
+            />
+          </div>
+          <div className="p-4">
+            <RadioButton
+              id={WithLogin.WithoutLogin}
+              name="with-auth"
+              caption={t(WithLogin.WithoutLogin)}
+              onChange={handleWithLoginChange}
+              value={WithLogin.WithoutLogin}
+              checked={!isWithLogin}
+              disabled={isSignedIn}
+            />
+          </div>
+        </div>
+      )}
+    </div>
+  );
+};
+
+export const AuthField = () => {
+  const { t } = useTranslation(Translation.Common);
+  const dispatch = useAppDispatch();
+
+  const toolsetDetails = useAppSelector(ToolsetSelectors.selectToolsetDetails);
+  const { control, trigger } = useFormContext<ToolsetEditorForm>();
+  const [endpoint, transport, allowedTools] = useWatch<ToolsetEditorForm>({
```

Comment:
```suggestion
  const [endpoint, transport, allowedTools] = useWatch({
```

Just remove it to get proper types
and then just remove type assertion on line 223